File: Scadex.Model/Dtos/Camera/Queries/StreamTokenDto.cs
using Scadex.Core.Model;

namespace Scadex.Model.Dtos.Camera.Queries;

public class StreamTokenDto : IDto
{
    /// <summary> 
    /// Tarayicinin SDP teklifini gonderecegi WHEP adresi. <para/> 
    /// <c>WebRtcPublicBaseUrl</c>'den üretilir => <b>"{WebRtcPublicBaseUrl}/{pathName}/whep"</b>
    /// </summary>
    public string WhepUrl { get; set; } = null!;
    /// <summary>
    /// Istemci token bilgisini <c>Authorization: Basic base64("token:" + token)</c> olarak gonderir; Medya gecidi(MediaMTX) de dogrulamak icin bize geri sorar.
    /// </summary>
    public string Token { get; set; } = null!;
    public DateTime ExpirationUtc { get; set; }
}

File: Scadex.WebUI/src/models/camera/queries/streamTokenDto.ts
/**
 * Ayna: Scadex.Model/Dtos/Camera/Queries/StreamTokenDto.cs
 * Sözleşme: docs/api-contract/11-camera.md
 */
export interface StreamTokenDto {
  /**
   * Tarayıcının SDP teklifini göndereceği WHEP adresi.
   *
   * Sunucunun değil TARAYICININ ulaşacağı adres. Bağlantı kurulamıyorsa ilk
   * bakılacak yer sunucudaki `Mediamtx:WebRtcPublicBaseUrl` ayarıdır —
   * `127.0.0.1` yazılıysa yalnızca sunucunun kendi tarayıcısında çalışır.
   */
  whepUrl: string;

  /**
   * Opak bilet. `Authorization: Basic base64("ticket:" + token)` olarak
   * gönderilir — kullanıcı adı kısmı sabit `ticket` dizesidir, alanın adı değil.
   *
   * Yola bağlıdır: A kamerası için alınan bilet B'nin adresinde çalışmaz.
   */
  token: string;

  /**
   * Biletin son geçerlilik anı (ISO-8601 UTC).
   *
   * Bağlantı koptuğunda eski bilet SAKLANMAZ; yeniden bağlanma her denemede
   * yenisini ister (bkz. `lib/camera/stream-session.ts`).
   */
  expirationUtc: string;
}
